e2e-tests/functions

```diff
@@ -572,6 +628,57 @@ retry() {
     done
 }
 
+start_vault() {
+    name=${1:-vault-service}
+    create_namespace "$name"
+    deploy_helm "$name"
+    helm delete --purge "$name" || :
+    
+    desc "install Vault $name"
+    helm install --name $name --namespace "$name" \
+        --set dataStorage.enabled=false \
+        "$src_dir/vendor/vault"
+
+    set +o xtrace
+    retry=0
+    echo -n pod/$name-0
+    until kubectl get pod/$name-0 -o 'jsonpath={.status.containerStatuses[0].state}' 2>/dev/null | grep 'running'; do
+        echo -n .
+        sleep 1
+        let retry+=1
+        if [ "$retry" -ge 480 ]; then
+            kubectl_bin describe pod/$name-0
+            kubectl_bin logs $name-0
+            echo max retry count "$retry" reached. something went wrong with vault
+            exit 1
+        fi
+    done
+    set -o xtrace
+}
+
+init_vault() {
+    name=${1:-vault-service}
+    kubectl exec -it pod/$name-0 -- vault operator init -tls-skip-verify -key-shares=1 -key-threshold=1 -format=json >"$tmp_dir/$name"
+    unsealKey=$(jq -r ".unseal_keys_b64[]" <"$tmp_dir/$name")
+    token=$(jq -r ".root_token" <"$tmp_dir/$name")
+    ip="$name.$name.svc.cluster.local"
+
+    kubectl exec -it pod/$name-0 -- vault operator unseal -tls-skip-verify "$unsealKey"
+    kubectl exec -it pod/$name-0 -- \
+        sh -c "export VAULT_ADDR=http://$ip:8200 && export VAULT_TOKEN=$token && export VAULT_LOG_LEVEL=trace \
+                && vault secrets enable --version=1 -tls-skip-verify -path=secret kv \
+                && vault audit enable file file_path=/vault/vault-audit.log"
+    sleep 10
+
+    #swich to test namestpace
+    kubectl_bin config set-context $(kubectl_bin config current-context) --namespace="$namespace"
+    cat "$conf_dir/vault-secret.yaml" |
+        sed -e "s/#token/$token/" |
+        sed -e "s/#vault_url/http:\/\/$ip:8200/" |
```

Comment:
I believe we must support https also.
yes, it will require to fetch CA and embed into a secret in somehow